spec/lita/handlers/stacker_spec.rb

```diff
@@ -185,6 +204,76 @@
     end
   end
 
+  describe '#lifo_requeue' do
+    let(:command) { 'restack' }
+
+    it_behaves_like 'a private chat'
+
+    context 'when the message is in a room' do
+      include_context 'in a room'
+
+      it_behaves_like 'it clears old stacks'
```

Comment:
Do you want to cover the transition case, where a single username might show up multiple times in the queue, and determine proper behavior there?  Or just go for a clean break instead?

Reply:
ohh, switching to sorted sets means that all stacks would be cleared anyway, is that right?

Reply:
The module to upgrade from lists to sorted sets [explicitly covers that case](https://github.com/change/lita-stacker/blob/71809d5b08698289ef36a68ba31fca3b4f500155/lib/lita/handlers/stacker/upgrade/sorted_sets.rb#L34) ([tested here](https://github.com/change/lita-stacker/blob/71809d5b08698289ef36a68ba31fca3b4f500155/spec/lita/handlers/stacker/upgrade/sorted_sets_spec.rb#L36-L39), perhaps that spec should read “…maintaining the correct order.”) because, yes. zsets only allow one instance of a given lexical value.

Reply:
…Which, I just realized, means that the new `stack` command can just add a user with a new score and the zset will handle everything else.